src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -399,6 +409,30 @@ bool CanApplyChange(DocumentId documentId)
             }
         }
 
+        private void NotifyRefactorChanges(IEnumerable<DocumentId> changedDocumentIds, CodeAnalysis.Solution oldSolution, CancellationToken cancellationToken = default)
+        {
+            // Without any services to notify there's no need to dig through the documents
+            if (!_refactorNotifyServices.Any())
+            {
+                return;
+            }
+
+            var changedSymbols = RenameSymbolAnnotation.GatherChangedSymbolsInDocumentsAsync(changedDocumentIds, CurrentSolution, oldSolution, cancellationToken).WaitAndGetResult(cancellationToken);
+            foreach (var changedSymbolPair in changedSymbols)
+            {
+                var oldSymbol = changedSymbolPair.Key;
+                var newSymbol = changedSymbolPair.Value;
+
+                foreach (var refactorNotifyService in _refactorNotifyServices)
+                {
+                    if (refactorNotifyService.Value.TryOnBeforeGlobalSymbolRenamed(this, changedDocumentIds, oldSymbol, newSymbol.ToDisplayString(), throwOnFailure: false))
+                    {
+                        refactorNotifyService.Value.TryOnAfterGlobalSymbolRenamed(this, changedDocumentIds, oldSymbol, newSymbol.ToDisplayString(), throwOnFailure: false);
```

Comment:
1. same comment as in the test version.  this logic seems bizarre to me.  what's the point of Before/After when we're not actually doing anything in between?